impeller/core/formats.h

```diff
@@ -293,6 +293,13 @@ constexpr const char* TextureUsageToString(TextureUsage usage) {
 
 std::string TextureUsageMaskToString(TextureUsageMask mask);
 
+enum class UsageHint {
```

Comment:
`ResourceUsageHint` perhaps? This is a bit ambiguous.

Reply:
Maybe rather than a hint it should be more explict as "ResourceUsage", because I think we should also adjust the usage flags to see if that helps. For example ResourceUsage.stagingBuffer should only be a transferSrc, and doesn't need random access, et cetera.

Reply:
we can have a "general" category that retains the current set of features